api/client/credentials.go

```diff
@@ -9,36 +9,74 @@ import (
 // GetCredentials loads the user credentials from a credentials store.
 // The store is determined by the config file settings.
 func GetCredentials(c *configfile.ConfigFile, serverAddress string) (types.AuthConfig, error) {
-	s := LoadCredentialsStore(c)
+	s := LoadCredentialsStore(c, serverAddress)
 	return s.Get(serverAddress)
 }
 
 // GetAllCredentials loads all credentials from a credentials store.
 // The store is determined by the config file settings.
 func GetAllCredentials(c *configfile.ConfigFile) (map[string]types.AuthConfig, error) {
-	s := LoadCredentialsStore(c)
-	return s.GetAll()
+	ret := make(map[string]types.AuthConfig)
+	// Add the AuthConfigs stored in registry-specific helpers.
+	for server := range c.CredentialHelpers {
+		s := LoadCredentialsStore(c, server)
+		if err := addAllAuths(s, ret); err != nil {
+			return nil, err
+		}
+	}
+	// Add the AuthConfigs in the default store.
+	s := LoadCredentialsStore(c, "")
+	if err := addAllAuths(s, ret); err != nil {
+		return nil, err
+	}
+	return ret, nil
+}
+
+// addAllAuths adds all of the AuthConfigs stored in s to allAuths.
+func addAllAuths(s credentials.Store, allAuths map[string]types.AuthConfig) error {
+	newAuths, err := s.GetAll()
```

Comment:
I don't think this is going to work as expected, in fact not even sure it works correctly right now. The native store uses the `AuthConfigs` to figure out what to fetch from the store and adds a config for each. This means every call to `GetAll` for each helper will return the full list of auths, some having empty values and the last value will win.